setup.cfg

```diff
@@ -38,7 +38,7 @@ install_requires=
     readme_renderer >= 21.0
     requests >= 2.20
     requests-toolbelt >= 0.8.0, != 0.9.0
-    setuptools >= 0.7.0
+    packaging
```

Comment:
We have minimum supported versions elsewhere to make supporting this easier, especially if re-distributed by linux distributions. Can we declare a bottom here for the inevitable case of a distro providing too old a version and us needing a newer version?

Reply:
Let's see if I can figure out how to do that.

I created an environment with `nox --install-only -e tests`, which created `.nox/tests-3-9/` then ran bisect:

```
packaging main $ git bisect start main 14.0
Bisecting: 230 revisions left to test after this (roughly 8 steps)
[08c1e52b3fef534f4132c107c54681979b52355a] Merge pull request #174 from asottile/fixes_pytest_5
packaging HEAD $ git bisect run .nox/tests-3-9/bin/python -c "from packaging.utils import canonicalize_name"
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 114 revisions left to test after this (roughly 7 steps)
[c250bf4d48ac4afc86f85f65c8cc039ad2a90005] Merge pull request #227 from pypa/fix-workflows
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 57 revisions left to test after this (roughly 6 steps)
[2a883ca86300bab76ad6296860797907b3432871] Avoid two function calls per comparison (#303)
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 28 revisions left to test after this (roughly 5 steps)
[69855f324f58d7a421fc1791b7794dfa80cf519d] Update CHANGELOG to reflect that 20.6 was yanked
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 14 revisions left to test after this (roughly 4 steps)
[1b26f8455d8c886cbfd37839eafab3373d1809f2] Support "macosx_10_9_universal2" wheels (#380)
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 7 revisions left to test after this (roughly 3 steps)
[c8a2f0b234251518319bff900bba4d6a8a7690a1] Fix type hints for Version.post and Version.dev (#393)
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 3 revisions left to test after this (roughly 2 steps)
[ae891fd74d6dd4c6063bb04f2faeadaac6fc6313] Clean up and improve typing (#378)
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 1 revision left to test after this (roughly 1 step)
[e0659c7dfb55b43aafb10a42d9950cc36423ae27] Use typing alias UnparsedVersion (#398)
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[bcd68bca7cb3d855e98c1515349306fdc8647f3d] Tighten the return type of canonicalize_version() (#402)
running .nox/tests-3-9/bin/python -c from packaging.utils import canonicalize_name
320a305b4d3066071aafa921b4641a68ede2c8ae is the first bad commit
commit 320a305b4d3066071aafa921b4641a68ede2c8ae
Author: Blazej Michalik <MrMino@users.noreply.github.com>
Date:   Thu Feb 11 00:07:23 2021 +0100

    Proper keyword-only "warn" argument in packaging.tags (#403)

 packaging/tags.py  | 27 ++++++---------------------
 tests/test_tags.py | 26 --------------------------
 2 files changed, 6 insertions(+), 47 deletions(-)
bisect run success
```

That implicated pypa/packaging@320a305b4d3066071aafa921b4641a68ede2c8ae as the latest commit that doesn't support the function.

Unfortunately, that was committed 19 days ago, so isn't what we're after.

[The changelog](https://pypi.org/project/packaging/) also isn't helpful. The only mention of that function is in adding type hints for it.

But checking out 14.0, I see that the utils module doesn't exist there, so it was added some time in the interim. The changelog doesn't even mention the utils module :(

So I guess I need to build a routine to examine all the versions on pypi to see which ones work.

Reply:
I figured out a xonsh routine to extract the min version:

```
packaging main $ def supported(tag):
................     git checkout -q @(tag)
................     proc = !(.nox/tests-3-9/bin/python -c "from packaging.utils import canonicalize_name" 2> /dev/null)
................     return not proc.rtn
................
................ tags = list(filter(supported, $(git tags).split()))
................ print(min(tags))
................
16.2
```

Reply:
Fixed in 96ff303.